R/class-plotdata.R

```diff
@@ -40,6 +41,16 @@ newPlotdata <- function(.dt = data.table(),
   lon <- veupathUtils::findColNamesFromPlotRef(variables, 'longitude')
 
   isEvil <- ifelse(evilMode %in% c('allVariables', 'strataVariables'), TRUE, FALSE)
+  collectionVarMetadata <- veupathUtils::findCollectionVariableMetadata(variables)
+  isOverlayCollection <- ifelse(is.null(collectionVarMetadata), FALSE, ifelse(collectionVarMetadata@plotReference@value %in% c('overlay'), TRUE, FALSE))
+
+  if (!is.null(group) && !isOverlayCollection) {
+    groupNeedsOverlayValues <- length(data.table::uniqueN(.dt[[group]])) > 8 && useGradientColorscale == FALSE
+    if (is.null(overlayValues) && groupNeedsOverlayValues) {
+      stop("Must provide overlay values of interest for high cardinality or continuous overlay variables.")
+    }
```

Comment:
add a check to ensure all overlayValues are actually values of the varaible?

Reply:
so i was thinking its possible we have a vocabulary like x,y,z and we request all values bc there are fewer than 8 or something, but in this particular subset we dont see all three values. so i think any type of validation would be hard.

Reply:
its possible the data service could try to validate before passing work to R.. like compare the values its passed to the known referenceMetadata

Reply:
yeah i didn't think about that the subset might not have the desired values. Instead of a validation, maybe it'd be enough to just log any desired values that didn't appear in dt. For a non-eda user, this would be good info, and in case we have any trouble in the eda it'd be easier to see what happened in the logs.